### DISCUSSION TITLE
What's the best way to debug 422s from webhooks?

### DISCUSSION THREAD

**Discussion by thomasmost** [tokio-rs/axum#2200] [Q&A]
I've clearly messed up my struct definition for the expected body of a certain request, but the 422 doesn't contain any additional information (in general, serde Deserialization errors leave something to be desired IMO)

So what's the best way of debugging this? Change the payload to `axum::Json<serde_json::Value>` and manually log it and then deserialize it in code? **Or is there a way to flag the endpoint to log raw output on a 422?**

```rs
// example of endpoint handler signature
pub async fn receive_event(
  State(pool): State<Pool>,
  payload: axum::Json<SomeEventBody>,
) -> StatusCode {
...
}
```

**Comment by davidpdrsn** [comment]
Try https://docs.rs/axum/0.6.20/axum/extract/index.html#logging-rejections

**Comment by thomasmost** [comment]
Thank you — this looks promising! I'll need to dig into how/when to enable this in my Cloud Run instances
